### DISCUSSION TITLE
Calmcode Course: what should people learn?

### DISCUSSION THREAD

**Discussion by koaning** [mkdocs/mkdocs#3665] [Ideas]
Mkdocs-Material is easy to learn (feature!) but the course that I maintain on [calmcode](https://calmcode.io/course/mkdocs/intro-to-mkdocs) is up for a revamp so I'm looking for things that may be less than obvious that might be good to teach. The main thing that comes to mind is "how to customise the landing page for something fancy" or maybe "how to deploy mkdocs with a bit of auth in front of it". 

I'll gladly hear suggestions for other ideas though. Especially if certain topics seem to be considered complex by the community.

**Comment by pawamoy** [comment]
Might be an ambitious one, but a remark we see often is that MkDocs offers a less complete experience than alternatives (Sphinx/Quarto), or at least a not as straight-forward experience. I had in mind that we could offer a "complete MkDocs experience" in one form or another (list of plugins, guide, meta-package, docker image, etc.) which would help people get the same level of completeness with MkDocs as these alternatives.

Completeness includes:

- versioning (mike is the most popular solution)
- generating the docs in other formats than HTML (useful for system packagers) like PDF, man pages, e-pub/mobi/other e-reader formats (https://pypi.org/project/mkdocs-exporter/ seems great for PDF, I myself have developed https://github.com/pawamoy/mkdocs-manpage for manpages)
- API refs (mkdocstrings, but there are alternatives)
- CLI refs (there are some specific packages like mkdocs-typer, though I usually handle it with a custom script and markdown-exec, example in PDM here https://github.com/pdm-project/pdm/blob/main/docs/reference/cli.md)
- ?

It would be amazing to get a Calmcode Course on that, but I understand it's a lot to unpack, and a lot of work. Feel free to pick any item individually :slightly_smiling_face:

**Reply by koaning** [reply]
I will probably and up with a final lesson where I just share a sane starting project with config and all that folks may copy. This will be opinionated, but I can imagine having something like that for "most Python projects" would already help out a ton.

**Comment by squidfunk** [comment]
Just a side note:

> how to customise the landing page for something fancy

Easier landing page creation is on our list as part of the rebrush/redesign we're going to do later this year.

**Reply by koaning** [reply]
Ah, good to know. I'm reminded why calmcode does many smaller videos as a course. That way, its much easier to replace bits when an update happens 🙂
